oci/pkg/oci/bundle.go

```diff
@@ -0,0 +1,156 @@
+package oci
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"reflect"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"gopkg.in/yaml.v2"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+func init() {
+	// Because we are using the K8s deserializer, we need to add Tekton's types to it.
+	schemeBuilder := runtime.NewSchemeBuilder(v1alpha1.AddToScheme)
+	schemeBuilder.AddToScheme(scheme.Scheme)
+}
+
+// ParsedTektonResource represents a full Tekton task, pipeline, etc that has been read in from the user along with
+// metadata about the resource.
+type ParsedTektonResource struct {
+	Name     string
+	Kind     *schema.GroupVersionKind
+	Contents string
+	Object   runtime.Object
+}
+
+// ReadPaths will recursively search each file path for Tekton resources and return the parsed specs or an error.
+func ReadPaths(filePaths []string) ([]ParsedTektonResource, error) {
```

Comment:
It looks like a lot of these methods are only called from within this package, so they don't have to be exported.

Reply:
I believe ReadPaths and ParsedTektonResource are the only exported members and they are referenced outside of the `oci` package inside of the `action` package. Perhaps there is a case to be made for re-organizing but this tends to follow a cobra cli layout as far as I can tell.